src/mame/sega/m1comm.cpp

```diff
@@ -581,25 +593,22 @@ int m1comm_device::read_frame(int dataSize)
 					togo -= recv;
 					offset += recv;
 				}
-				else if (!filerr && recv == 0)
+				if ((!filerr && recv == 0) || (filerr && std::errc::operation_would_block != filerr))
 				{
 					togo = 0;
 				}
 			}
 		}
 	}
-	else if (!filerr && recv == 0)
+	if ((!filerr && recv == 0) || (filerr && std::errc::operation_would_block != filerr))
 	{
+		osd_printf_verbose("M1COMM: rx connection error\n");
```

Comment:
It would be helpful to log the actual error.

Reply:
I now log the errors.